subprojects/analysis-kotlin-api/src/testFixtures/kotlin/org/jetbrains/dokka/analysis/test/api/configuration/TestDokkaConfigurationBuilder.kt

```diff
@@ -0,0 +1,143 @@
+
+import org.jetbrains.dokka.analysis.test.api.TestProject
+import org.jetbrains.dokka.analysis.test.api.util.AnalysisTestDslMarker
+
+/**
+ * A builder for [TestDokkaConfiguration] that contains base options.
+ *
+ * Implementations can override the properties to modify the resulting test
+ * data or prohibit setting certain properties.
+ */
+@AnalysisTestDslMarker
+abstract class BaseTestDokkaConfigurationBuilder {
+
+    /**
+     * @see TestDokkaConfiguration.moduleName
+     */
+    abstract var moduleName: String
+
+
+    /**
+     * @see TestDokkaConfiguration.includes
+     */
+    open var includes: Set<String> = emptySet()
+
+    /**
+     * Verifies that this source set configuration is valid. For example,
+     * it should check that all file paths are set in the expected format.
+     *
+     * Must be invoked manually as part of [TestProject.verify].
+     */
+    open fun verify() {
+        includes.forEach {
+            verifyFilePathStartsWithSlash("includes", it)
+            verifyFileExtension("includes", it, ".md")
+        }
+    }
+
+    abstract fun build(): TestDokkaConfiguration
+
+    override fun toString(): String {
+        return "BaseTestDokkaConfigurationBuilder(moduleName='$moduleName', includes=$includes)"
+    }
+}
+
+/**
+ * A builder for [TestDokkaSourceSet] that contains base options.
+ *
+ * Implementations can override the properties to modify the resulting test
+ * data or prohibit setting certain properties.
+ */
+@AnalysisTestDslMarker
+abstract class BaseTestDokkaSourceSetBuilder {
+
+    /**
+     * Directories **additional** to the default source roots.
+     *
+     * @see TestDokkaSourceSet.sourceRoots
+     */
+    open var additionalSourceRoots: Set<String> = emptySet()
+
+    /**
+     * JARs **additional** to the default classpath.
+     *
+     * @see TestDokkaSourceSet.classpath
+     */
+    open var additionalClasspath: Set<String> = emptySet()
+
+    /**
+     * @see TestDokkaSourceSet.includes
+     */
+    open var includes: Set<String> = emptySet()
+
+    /**
+     * @see TestDokkaSourceSet.samples
+     */
+    open var samples: Set<String> = emptySet()
+
+    /**
+     * @see TestDokkaSourceSet.languageVersion
+     */
+    open var languageVersion: String? = null
+
+    /**
+     * @see TestDokkaSourceSet.apiVersion
+     */
+    open var apiVersion: String? = null
+
+    /**
+     * Verifies that this source set configuration is valid. For example,
+     * it should check that all file paths are set in the expected format.
+     *
+     * Must be invoked manually during the verification of the
+     * higher-level [BaseTestDokkaConfigurationBuilder.verify].
+     */
+    open fun verify() {
+        additionalSourceRoots.forEach {
+            verifyFilePathStartsWithSlash("additionalSourceRoots", it)
+        }
+        additionalClasspath.forEach {
+            verifyFileExtension("additionalClasspath", it, ".jar") // should it include .klib?
```

Comment:
> It can be a list of JARs, KLIBs, folders with .class files.

Reply:
I'll fix the comment :+1:. Let's change the actual check once we have the first related tests and are sure that they work.

Reply:
Oh, sorry, missed the reply.

Not sure what you mean exactly. The only place where `kotlin/jvm/Strictfp.class` is used is to find the `.jar` file that contains it - passing the `.jar` file here works, yeah. But for now there's no need and there are no tests that pass individual class files

Reply:
My bad. I thought `jvmStdlibPath` was an unpacked folder.

> But for now there's no need and there are no tests that pass individual class files

Indeed, the stdlib jar is only used in the classpath.

> // this check can be extended to accept .klib, .class or other files

I'm not sure Dokka's analysis can work with individual `.class` files. But it should work with directories containing `.class` files, e.g. `someProject/build/classes/`.